### PR TITLE
Add missing discovery fields "Integration" and "KubeAppDiscovery" to the file config

### PR DESCRIPTION
**Pull request by AntonAM**
These discovery fields were added to AWS matchers and could be set through dynamic config, but were missing from the file config. This PR adds those fields. Also we enable dynamic watching for non-integration fetchers and also refactor kube integration vs non integration watchers to be a bit more clear.

Changelog: Add missing discovery AWS matchers fields "Integration" and "KubeAppDiscovery" to the file configuration.

### CODE REVIEW

File: lib/srv/discovery/fetchers/eks.go [hunk 1]
```diff
@@ -170,6 +165,10 @@ func (a *eksFetcher) getClient(ctx context.Context) (eksiface.EKSAPI, error) {
 	return a.client, nil
 }
 
+func (a *eksFetcher) GetIntegration() string {
```
Comment (marcoandredinis) [inline lib/srv/discovery/fetchers/eks.go:168]: Missing godoc

File: lib/srv/discovery/kube_integration_watcher.go [hunk 2]
```diff
@@ -220,16 +226,48 @@ func (s *Server) getKubeAgentVersion(releaseChannels automaticupgrades.Channels)
...
+	filter := func(fetcher common.Fetcher) bool {
+		f, ok := fetcher.(IntegrationFetcher)
+		if !ok {
+			// If fetcher doesn't provide integration field, return true if non-integration fetchers were requested.
+			return !integration
+		}
+		// If fetcher provides integration field, return true if integration fetchers were requested and integration
+		// field actually has a value.
+		return integration == (len(f.GetIntegration()) > 0)
+	}
```
Comment (marcoandredinis) [inline lib/srv/discovery/kube_integration_watcher.go:246]: I'm not questioning the correctness of this function, but it is not clear at first sight that it is indeed doing the right thing.

I think creating two distinct functions would be easier to read
```go

	filterIntegrationFetchers := func(fetcher common.Fetcher) bool {
		f, ok := fetcher.(IntegrationFetcher)
		if !ok {
			return false
		}

		return f.GetIntegration() != ""
	}

	filterNonIntegrationFetchers := func(fetcher common.Fetcher) bool {
		f, ok := fetcher.(IntegrationFetcher)
		if !ok {
			return true
		}

		return f.GetIntegration() == ""
	}
```

Creating `GetIntegration` in the `common.Fetcher` interface would help as well.

**Comment by marcoandredinis** [comment]
Btw, I've tested the Integration field and it is working as expected :+1:

File: lib/srv/discovery/kube_integration_watcher.go (hunk 2 above)
Comment (AntonAM) [reply lib/srv/discovery/kube_integration_watcher.go]: Refactored 👍 https://github.com/gravitational/teleport/pull/40155/commits/3eeed03f51159738b9501a62195c0a5a0c8682dd

**Comment by AntonAM** [comment]
@smallinsky friendly ping :)

**Review by marcoandredinis — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Pull request merged**